### PR TITLE
Monitoring dashboards: Add text filter to the Dashboards dropdown and the variable dropdowns

### PR DESCRIPTION
**Pull request by kyoto**
Jira ticket: https://issues.redhat.com/browse/MON-1521

Also moves `TimespanDropdown` to its own file since it no longer uses `VariableDropdown`.

screenshot

### CODE REVIEW

**Comment by sichvoge** [comment]
Does this work for the "Dashboards" drop down as well?

**Comment by kyoto** [comment]
@sichvoge I hit a problem with the dashboards dropdown, so decided to leave that for a separate PR.

File: frontend/public/components/monitoring/dashboards/index.tsx [hunk 1]
```diff
@@ -105,7 +112,27 @@ const VariableDropdown: React.FC<VariableDropdownProps> = ({
...
+    if (isExpanded) {
+      open();
+    } else {
+      close();
+      setFilterText(undefined);
+    }
+  };
+
+  const filteredItems = filterText
+    ? _.pickBy(items, (v) => v.toLowerCase().includes(filterText))
```
Comment (spadgett) [inline frontend/public/components/monitoring/dashboards/index.tsx:134]: Worth a comment explaining the `filterText` is lower-cased before it's saved in state.

**Review by spadgett — APPROVED** [review]
/lgtm
/hold for approvals

**Comment by kyoto** [comment]
@spadgett I ended up extending and reworking this
- Rebased
- Addressed your feedback comment
- Added the text filter for the Dashboards dropdown too
- Refactored the `isError` case to also use a `Select`

**Comment by lihongyan1** [comment]
Filed bug 
https://bugzilla.redhat.com/show_bug.cgi?id=1984365

**Comment by lihongyan1** [comment]
/label qe-approved
The bug is not critical and not very related to the PR

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by kyoto** [comment]
/assign @sferich888
for px- approval

**Comment by spadgett** [comment]
/label px-approved
/hold cancel

@kyoto Needs rebase :/

**Comment by kyoto** [comment]
@spadgett Rebased

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**